tests/java/iceberg-rest/build.gradle

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2024 Tabular Technologies Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+  id 'java'
+  id 'com.github.johnrengelman.shadow' version '8.1.1'
+}
+
+repositories {
+  mavenCentral()
+}
+
+ext {
+  icebergVersion = '1.6.0'
+  hadoopVersion = '3.3.6'
+  awsJavaSdkVersion = '1.12.367'
+}
+
+dependencies {
+  // Needed for HadoopCatalog's s3a access at runtime
+  // while technically this is just a runtime dependency there is
+  // some pollution in the shaded jar that pulls in duplicate
+  // implementations of S3ABlockOutputStream. Making it a compile
+  // time dependency works around the problem by probably loading
+  // it ahead of other implementations. This is a dirty hack but
+  // probably ok since this is just for tests and it is mind numbing
+  // to debug hadoop dependency mess :'(
+  implementation "org.apache.hadoop:hadoop-aws:${hadoopVersion}"
+  runtimeOnly "com.amazonaws:aws-java-sdk-bundle:${awsJavaSdkVersion}"
```

Comment:
changed by me: add hadoop aws dependencies to the shaded jar to get HadoopCatalog working.